### PR TITLE
[move] Consolidate source/dep logic 

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- We have layered on some notions of dependencies/source targets because of the package system. 
- I have created an enum with 3 variants, but I kind of hate the name. So open to suggestions here

## Test plan 

- Internal change 

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: external-crates/move/crates/move-compiler/src/expansion/ast.rs [hunk 1]
```diff
@@ -99,6 +99,18 @@ pub type Attributes = UniqueMap<Spanned<KnownAttribute>, Attribute>;
...
+#[derive(Debug, Clone, Copy)]
+/// Specifies a source target or dependency
+pub enum TargetKind {
+    /// A source module
+    Source,
+    /// A dependency where bytecode is being generated
+    DependencyBeingCompiled,
+    /// A dependency only used for linking.
+    /// No bytecode or CompiledModules are generated
+    DependencyBeingLinked,
```
Comment (tnowacki) [inline external-crates/move/crates/move-compiler/src/expansion/ast.rs:111]: I kind of hate these names, open to suggestions

Comment (cgswords) [reply external-crates/move/crates/move-compiler/src/expansion/ast.rs:111]: I agree, these names don't communicate what is going on very well.

A few suggestions:

1) 
```
pub enum TargetKind {
  Dependency { from_source bool },
  RootProject // or similar
}
```

2) 
```
pub enum TargetKind {
  SourceDependency,
  BytecodeDependency,
  Source
}
```

I also asked ChatGPT:

```
enum PackageType {
    MainProject,
    SourceDependency,
    BytecodeDependency,
}
```

File: external-crates/move/crates/move-compiler/src/sui_mode/id_leak.rs [hunk 2]
```diff
@@ -105,7 +105,7 @@ impl SimpleAbsIntConstructor for IDLeakVerifier {
             // Skip if not sui
             return None;
         }
-        if config.is_dependency || !mdef.is_source_module {
+        if !matches!(mdef.target_kind, TargetKind::Source) {
             // Skip non-source, dependency modules
```
Comment (cgswords) [inline external-crates/move/crates/move-compiler/src/sui_mode/id_leak.rs:109]:
```suggestion
            // Skip non-source and dependency modules
```

File: external-crates/move/crates/move-compiler/src/sui_mode/typing.rs [hunk 3]
```diff
@@ -114,7 +114,7 @@ impl<'a> TypingVisitorContext for Context<'a> {
             // Skip if not sui
             return true;
         }
-        if config.is_dependency || !mdef.is_source_module {
+        if !matches!(mdef.target_kind, TargetKind::Source) {
             // Skip non-source, dependency modules
```
Comment (cgswords) [inline external-crates/move/crates/move-compiler/src/sui_mode/typing.rs:118]:
```suggestion
            // Skip non-source and dependency modules
```

**Review by cgswords — APPROVED** [review]
I agree the names are _very_ confusing as they are, so approving assuming they will be fixed before landing. 🙏

File: external-crates/move/crates/move-compiler/src/expansion/ast.rs (hunk 1 above)
Comment (tnowacki) [reply external-crates/move/crates/move-compiler/src/expansion/ast.rs:111]: My issue with "BytecodeDependency" is it makes it feel like it was from an `.mv` file

**Pull request merged**